lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -1365,14 +1365,31 @@ SPIRVValue *LLVMToSPIRV::transValueWithoutDecoration(Value *V,
     return mapValue(V, BI);
   }
 
-  if (SelectInst *Sel = dyn_cast<SelectInst>(V))
-    return mapValue(
-        V,
-        BM->addSelectInst(
-            transValue(Sel->getCondition(), BB),
-            transValue(Sel->getTrueValue(), BB, true, FuncTransMode::Pointer),
-            transValue(Sel->getFalseValue(), BB, true, FuncTransMode::Pointer),
-            BB));
+  if (SelectInst *Sel = dyn_cast<SelectInst>(V)) {
+    auto CV = transValue(Sel->getCondition(), BB);
+    auto TV = transValue(Sel->getTrueValue(), BB, true, FuncTransMode::Pointer);
+
+    if (!CV->getType()->isTypeVector() && TV->getType()->isTypeVector()) {
```

Comment:
I think it would be nice to have a comment with short description why we are doing this.